### PR TITLE
Add a scroll to bottom on message sent setting

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#16548](https://github.com/vector-im/element-web/issues/16548)

### CODE REVIEW

**Comment by t3chguy** [comment]
18 minutes, impressive :P

**Comment by nunoperalta** [comment]
Hey! Thanks for this.
Is this setting stored in the database per user, or is it a Browser-level setting stored in a cookie or similar?

**Comment by SimonBrandner** [comment]
@nunoperalta, it's a user-level setting - it should get synced across devices

**Comment by nunoperalta** [comment]
> @nunoperalta, it's a user-level setting - it should get synced across devices

Perfect! Thank you very much!

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ export default class PreferencesUserSettingsTab extends React.Component {
         'showRedactions',
         'enableSyntaxHighlightLanguageDetection',
         'expandCodeByDefault',
+        `scrollToBottomOnMessageSent`,
```
Comment (turt2live) [inline src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js:51]: Matching quote styles in the area is important to us:
```suggestion
        'scrollToBottomOnMessageSent',
```

**Review by turt2live — APPROVED** [review]
code looks fine, though I do think this should be a product call for if we want this feature and/or setting.

**Comment by nunoperalta** [comment]
> I do think this should be a product call for if we want this feature and/or setting

Does implementing this feature cause any impact/problem to the system?

As explained in #16548, one of the main reasons (if not the only reason) my team moved to Riot/Element was that we can reply to several pending messages in a room, without having to constantly being pushed down and scroll back up.

This "trending feature" has never made any sense to us. We can't think of any reason we want to be pushed to the very bottom, when we reply to a message, "forcing" us to ignore everything else that the other person has sent.

Please, if this setting doesn't cause any harm to Element, this would be very beneficial to us.

**Comment by turt2live** [comment]
There's always maintenance burden and the potential for settings sprawl - if we can make do without a setting, we generally prefer that route. I'm honestly a little surprised that design/product wasn't involved on the original review given it's such a polarizing feature.

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js (hunk 1 above)
Comment (SimonBrandner) [reply src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js]: I've done edits to this file before but this time my brain just decided to do a weird mistake :joy:

**Comment by turt2live** [comment]
well, I've spent the last day thinking about this and tbh I think that if the original feature made it through without product review then this can too - it seems like a logical improvement given the change.

**Comment by nunoperalta** [comment]
Thank you for this. This seems to be working!

**Pull request merged**